src/python/base/errors.py

```diff
@@ -110,8 +110,12 @@ def __str__(self):
 class BadConfigError(Error):
   """Error thrown when configuration is bad."""
 
-  def __init__(self):
-    super(BadConfigError, self).__init__('Bad configuration.')
+  def __init__(self, config_dir=None):
```

Comment:
why is config_dir optional, lets make it there always, i see those are two call sites.

Reply:
Good catch. I added this for debugging and meant to revert it but might as well keep.